packages/react-components/react-alert/src/components/Alert/useAlert.tsx

```diff
@@ -0,0 +1,66 @@
+import * as React from 'react';
+
+import { Button } from '@fluentui/react-button';
+import {
+  CheckmarkCircle16Filled,
+  DismissCircle16Filled,
+  ErrorCircle16Filled,
+  Warning16Filled,
+} from '@fluentui/react-icons';
+import { getNativeElementProps, resolveShorthand } from '@fluentui/react-utilities';
+
+import type { AlertProps, AlertState } from './Alert.types';
+
+/**
+ * Create the state required to render Alert.
+ *
+ * The returned state can be modified with hooks such as useAlertStyles_unstable,
+ * before being passed to renderAlert_unstable.
+ *
+ * @param props - props from this instance of Alert
+ * @param ref - reference to root HTMLElement of Alert
+ */
+export const useAlert_unstable = (props: AlertProps, ref: React.Ref<HTMLElement>): AlertState => {
+  const { intent } = props;
+
+  /** Determine the icon to render based on the intent */
+  let iconToUse;
+  switch (intent) {
+    case 'success':
+      iconToUse = <CheckmarkCircle16Filled />;
+      break;
+    case 'error':
+      iconToUse = <DismissCircle16Filled />;
+      break;
+    case 'warning':
+      iconToUse = <Warning16Filled />;
+      break;
+    case 'info':
+      iconToUse = <ErrorCircle16Filled />;
+      break;
+  }
+
+  const icon = resolveShorthand(props.icon, {
+    defaultProps: {
+      children: iconToUse,
```

Comment:
```suggestion
      children: defaultIcon,
```

_nit_: I would use `defaultIcon` or something similar to indicate its behavior better (as this icon can be overridden by a user)

Reply:
Good idea, done